generators/argonaut-r4/generator.rb

```diff
@@ -139,10 +140,61 @@ def create_interaction_test(sequence, interaction)
   sequence[:tests] << interaction_test
 end
 
+def create_must_support_test(sequence)
+  test = {
+    tests_that: 'Demonstrates that the server can supply must supported elements',
+    index: sequence[:tests].length + 1,
+    link: 'https://build.fhir.org/ig/HL7/US-Core-R4/general-guidance.html/#must-support',
+    test_code: ''
+  }
+
+  extensions_list = []
+  sequence[:must_supports].select { |must_support| must_support[:type] == 'extension' }.each do |extension|
+    extensions_list << "'#{extension[:id]}': '#{extension[:url]}'"
+  end
+  if extensions_list.any?
+    test[:test_code] += %(
+        extensions_list = {
+          #{extensions_list.join(",\n          ")}
+        }
+        extensions_list.each do |id, url|
+          already_found = @instance.must_support_confirmed.include?(id.to_s)
+          element_found = already_found || @#{sequence[:resource].downcase}.extension.any? { |extension| extension.url == url }
+          skip "Could not find \#{id} in the provided resource" unless element_found
+          @instance.must_support_confirmed += "\#{id}," unless already_found
+        end
+)
+  end
+  elements_list = []
+  sequence[:must_supports].select { |must_support| must_support[:type] == 'element' }.each do |element|
+    element[:path] = element[:path].gsub('.class', '.local_class') # class is mapped to local_class in fhir_models
+    elements_list << "'#{element[:path]}'"
+  end
+
+  if elements_list.any?
+    test[:test_code] += %(
+        must_support_elements = [
+          #{elements_list.join(",\n          ")}
+        ]
+        must_support_elements.each do |path|
+          truncated_path = path.gsub('#{sequence[:resource]}.', '')
+          already_found = @instance.must_support_confirmed.include?(path)
+          element_found = already_found || can_resolve_path(@#{sequence[:resource].downcase}, truncated_path)
+          skip "Could not find \#{path} in the provided resource" unless element_found
+          @instance.must_support_confirmed += "\#{path}," unless already_found
+        end)
+  end
+
+  test[:test_code] += %(
+        @instance.save!)
+
+  sequence[:tests] << test
+end
+
 def create_resource_profile_test(sequence)
   test = {
     tests_that: "#{sequence[:resource]} resources associated with Patient conform to Argonaut profiles",
```

Comment:
Could you rename to 'US Core R4 profiles'?

Reply:
Done